### PR TITLE
[C][ModelPart] finished implementation

### PR DESCRIPTION
**Pull request by philbucher**
I started to implement the C-ModelPart

for me the main question is still how to do the return of multiple things (e.g. node from ModelPart or coordinates from Nodes)
[this](https://stackoverflow.com/a/11656585) is a nice explanation of the options, which way should we go @pooyan-dadvand ?

### CODE REVIEW

File: co_sim_io/c/co_sim_io_c_model_part.h [hunk 1]
```diff
@@ -55,24 +55,32 @@ typedef enum
...
+// ModelPart functions
 CoSimIO_ModelPart CoSimIO_CreateModelPart(const char* I_Name);
 
 int CoSimIO_FreeModelPart(CoSimIO_ModelPart I_ModelPart);
 
 const char* CoSimIO_ModelPart_Name(CoSimIO_ModelPart I_ModelPart);
 int CoSimIO_ModelPart_NumberOfNodes(CoSimIO_ModelPart I_ModelPart);
 int CoSimIO_ModelPart_NumberOfElements(CoSimIO_ModelPart I_ModelPart);
+// CoSimIO_Node* CoSimIO_ModelPart_Nodes(CoSimIO_ModelPart I_ModelPart);
+// CoSimIO_Element* CoSimIO_ModelPart_Elements(CoSimIO_ModelPart I_ModelPart);
```
Comment (philbucher) [inline co_sim_io/c/co_sim_io_c_model_part.h:83]: @pooyan-dadvand this is where I am not sure what would be better

Comment (pooyan-dadvand) [reply co_sim_io/c/co_sim_io_c_model_part.h:83]: This is correct. Arrays in c are returned by pointer to the first element. Using this one can use normal indexing our pointer jump to access to the members of the array. 

What is important is the memory management that here everything would belong to the CosimIO. Means that you should have this array also internally stored and do not create it inside your function.

Comment (pooyan-dadvand) [reply co_sim_io/c/co_sim_io_c_model_part.h:83]: I think that maybe a cleaner solution would be to provide only index based access to the entities like:
```
CoSimIO_Node* CoSimIO_ModelPart_Node_By_Index(CoSimIO_ModelPart I_ModelPart, int I_node_index);
```

**Comment by philbucher** [comment]
@pooyan-dadvand finally the C - `ModelPart` is ready and passes all the tests :tada:

**Review by pooyan-dadvand — APPROVED** [review]
I think the interface is minimal and clean. 👍

**Pull request merged**